src/app/shared/validators/min-max-validate.directive.ts

```diff
@@ -41,16 +50,22 @@ export class MinMaxValidateDirective implements Validator {
       return error;
     }
 
-    if (valueNumber < this.appMinValidate) {
+    if (valueNumber < this.min) {
       error[ErrorCode.min] = { value: value };
       return error;
     }
 
-    if (valueNumber > this.appMaxValidate) {
+    if (valueNumber > this.max) {
       error[ErrorCode.max] = { value: value };
       return error;
     }
 
     return null;
   }
+
+  private getSettings() {
+    this.min = this.appMinMaxValidate.min === undefined ? -Infinity : this.appMinMaxValidate.min;
+    this.max = this.appMinMaxValidate.max === undefined ? Infinity : this.appMinMaxValidate.max;
+    this.required = this.appMinMaxValidate.required === undefined ? false : this.appMinMaxValidate.required;
```

Comment:
Why do we need these as fields? This relies purely on side effects now.

Reply:
done